### PR TITLE
Remove hard coded version number; Generate at build time

### PR DESCRIPTION
**Pull request by DenverM80**

### CODE REVIEW

File: dsb-api/src/main/java/com/spectralogic/dsbrowser/api/services/BuildInfoService.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ *    CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ *    specific language governing permissions and limitations under the License.
+ *  ****************************************************************************
+ */
+
+package com.spectralogic.dsbrowser.api.services;
+
+public interface BuildInfoService {
+    String getBuildVersion();
+    String getBuildDate();
```
Comment (rpmoore) [inline dsb-api/src/main/java/com/spectralogic/dsbrowser/api/services/BuildInfoService.java]: Update to use the java 8 date object.

File: dsb-gui/build.gradle [hunk 2]
```diff
@@ -42,3 +45,19 @@ shadowJar {
...
+
+        File configFile = new File(sourceSets.main.output.resourcesDir, "/ds3_cli.properties")
+        configFile.withWriter{out ->
+            out.writeLine("version="+version)
+            out.writeLine("build.date="+new Date().toString())
+        }
+    }
+}
+
+shadowJar.dependsOn genConfigProperties
```
Comment (rpmoore) [inline dsb-gui/build.gradle]: This should happen on the jar task.  I'd verify that the jar task is called when the shadowJar is created, but it should be.

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/BuildInfoServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+    private final String buildVersion;
+    private final String buildDate;
+
+    public BuildInfoServiceImpl() {
+        final InputStream is = getClass().getClassLoader().getResourceAsStream(buildPropertiesFile);
+        final Properties buildProps = new Properties();
+        try {
+            buildProps.load(is);
+        } catch (IOException e) {
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/BuildInfoServiceImpl.java]: Missing final

Comment (RachelTucker) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/BuildInfoServiceImpl.java]: missing final

File: dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/BuildInfoServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+public class BuildInfoServiceImpl implements BuildInfoService{
+    private static final Logger LOG = LoggerFactory.getLogger(BuildInfoServiceImpl.class);
+
+    public static final String buildPropertiesFile = "ds3_cli.properties";
+
+    private final String buildVersion;
+    private final String buildDate;
+
+    public BuildInfoServiceImpl() {
+        final InputStream is = getClass().getClassLoader().getResourceAsStream(buildPropertiesFile);
```
Comment (rpmoore) [inline dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/BuildInfoServiceImpl.java]: Put this logic into it's own method.  Typically constructors should only be assigning things, not performing any logic or IO handling.

File: dsb-gui/build.gradle (hunk 2 above)
Comment (DenverM80) [inline dsb-gui/build.gradle]: there is no jar task..?

**Comment by DenverM80** [comment]
updated

**Pull request merged**